06-building-a-program-start-to-finish/solving-the-problem/output/inverted-index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * An inverted-index class
+ * @class
+ */
+class InvertedIndex {
+  /**
+  * class constructor
+  * @constructor
+  */
+  constructor() {
+    this.allIndex = {};
+    this.allFiles = {};
+    this.allLength = {};
+    this.allFilesTitle = [];
+    this.validatedFile = {};
+  }
+
+  /**
+   * Validate Method
+   * @param {Object} validateFile an object to be validated
+   * @returns {Object} file that has been validated
+   */
+  validateFileFunc(validateFile) {
+    this.validatedFile = validateFile;
+    if (validateFile.title !== undefined && validateFile.text !== undefined) {
+      return validateFile;
+    } return false;
+  }
+
+  /**
+   * Sanitizer
+   * @param {Object} BookObject recieves json object
+   * @return {Object} returns an Object of names
+   */
+  sanitizer(BookObject) {
+    Object.keys(BookObject).forEach((titles) => {
+      this.allFilesTitle.push(BookObject[titles].title);
+    });
+  }
+
+  /**
+   * Get all Indecies
+   * @param {string} bookName name of the individaul book
+   * @returns {Object} allIndicies returns all indexed file
+   */
+  getAllIndecies(bookName) {
+    if (bookName !== undefined) {
+      return this.allFiles[bookName];
+    } return false;
+  }
+
+  /**
+   * Gets Indexes
+   * @param {Object} BookObject receives json object
+   * @param {string} bookName of the books
+   * @return {object} returns an object of Indexes
+   */
+  createIndex(BookObject, bookName) {
+    this.sanitizer(BookObject);
+    this.allLength[bookName] = BookObject.length;
+    BookObject.forEach((document, position) => {
+      const words = document.text
+      .toLowerCase()
+      .match(/\w+/g);
+      words.forEach((word) => {
+        if (this.allIndex[word]) {
+          if (!this.allIndex[word][position]) {
+            this.allIndex[word][position] = true;
+          }
+        } else {
+          const wordIndex = {};
+          wordIndex[position] = true;
+          this.allIndex[word] = wordIndex;
+        }
+      });
+    });
+    this.allFiles[bookName] = this.allIndex;
+    this.allIndex = {};
+    return this.allFiles[bookName];
+  }
+
+  /**
+   * Search function
+   * @param {string} queryString word to Search
+   * @param {string} filterName book name to search
+   * @return {Object} searchResult
+   */
+  searchFiles(queryString, filterName) {
```

Comment:
I believe `filterName` is the name of the file you're searching through. Is there a way to name this better?